src/Plugin.php

```diff
@@ -240,6 +239,10 @@ private function loadTextdomain(): void
     public function saveOptionsHook(): void
     {
         if (PluginHelper::isGdprPluginAdminScreen()) {
+            if (PluginHelper::shouldExport()) {
+                $exporter = new Exporter();
+                $exporter->exportToJsonFile();
```

Comment:
would exit here instead in the Exporter Class, otherwise it looks you are continuing the process, after writing file to fs. Maybe we can rename the method, like getBinaryFileResponse()

Reply:
I don't write to the filesystem. It's a direct download. But i agree, not exiting here, does look weird